src/threshold_decryption.rs

```diff
@@ -115,22 +115,32 @@ impl<N: NodeIdT> ThresholdDecryption<N> {
     /// Sets the ciphertext, sends the decryption share, and tries to decrypt it.
     /// This must be called exactly once, with the same ciphertext in all participating nodes.
     /// If we have enough shares, outputs the plaintext.
-    pub fn set_ciphertext(&mut self, ct: Ciphertext) -> Result<Step<N>> {
+    pub fn set_message(&mut self, ct: Ciphertext) -> Result<()> {
         if self.ciphertext.is_some() {
             return Err(Error::MultipleInputs(Box::new(ct)));
         }
+        self.ciphertext = Some(ct);
+        Ok(())
+    }
+
+    /// Sends our signature shares, and if we have collected enough, returns the decrypted message.
+    /// Returns an empty step if the ciphertext hasn't been received yet.
+    pub fn sign(&mut self) -> Result<Step<N>> {
+        let ct = match self.ciphertext {
+            None => return Ok(Step::default()),
+            Some(ref ct) => ct.clone(),
+        };
+        let mut step = Step::default();
+        step.fault_log.extend(self.remove_invalid_shares());
         if !self.netinfo.is_validator() {
-            self.ciphertext = Some(ct);
-            return Ok(self.try_output()?);
+            step.extend(self.try_output()?);
+            return Ok(step);
         }
         let share = match self.netinfo.secret_key_share().decrypt_share(&ct) {
-            None => return Err(Error::InvalidCiphertext(Box::new(ct))),
+            None => return Err(Error::InvalidCiphertext(Box::new(ct.clone()))),
```

Comment:
I'd actually [`verify`](https://docs.rs/threshold_crypto/0.2.1/threshold_crypto/struct.Ciphertext.html#method.verify) this in `set_ciphertext` and use [`decrypt_share_no_verify`](https://docs.rs/threshold_crypto/0.2.1/threshold_crypto/struct.SecretKeyShare.html#method.decrypt_share_no_verify) here.